### ISSUE TITLE
Style loading on simulator appears to be problematic on M1 Macs

### ISSUE THREAD

**Issue by nvanfleet** [maplibre/maplibre-native#257]
Since I got an M1 Mac I started getting some intermittent issues related to the loading the style. Hooking into the network stack I see that we 
1. Hand the style URL to the SDK
2. The style successfully downloads
3. Logger message from Maplibre that it will load the map

Results:
1. The map just never loads and stays white. 
2. It seems like sometimes after many seconds it will load finally OR if you set a new style it will instantly load the first style (as seen in the logs) and then loads the new style successfully.

This has also seemingly broken unit tests in the portion where we're trying to load the style before executing tests. 

I should note that we sometimes use https URLs but at times (especially when testing) we're passing a fileURL.

**Comment by racer1988** [comment]
for me MapLibre on M1 crashes all the time on simulator due to OpenGL rendering failing.

**Comment by nvanfleet** [comment]
> for me MapLibre on M1 crashes all the time on simulator due to OpenGL rendering failing.

That's interesting I have not seen that at all using the latest release. From what I have seen it might be that the M1 loads styles from disk _so fast_ that it was not being caught by something we we plugging into the map as delegate. Though it should not technically have affected the style being displayed properly.

**Comment by roblabs** [comment]
The issue of a "slow loading map" is 100% repeatable when testing on a Simulator running on Apple Silicon.  No issue found on an Intel based Mac.  Both tests were using the default MapLibre `style.json`.

I did notice in testing on Apple Silicon that the loading of the map is slow when testing on a Simulator _from within Xcode_ (even with LLDB debugger off).  When I launched the app in the simulator _outside of Xcode_ on Apple Silicon, then the map loads within about the same time as on an Intel Mac.  

I know from the MetalANGLE testing & acceptance criteria that Xcode may be doing extra layers of "stuff" that does not occur in the Simulator.  See discussion at https://developer.apple.com/forums/thread/659856?answerId=679202022#679202022.
